[readline-unix] Build fails in Manifest mode using overlay triplets

Package: readline-unix[core]:arm92-buildroot-linux-dynamic -> 8.1#2

**Host Environment**

- Host: x64-linux
- Compiler: GNU 9.2.0
-    vcpkg-tool version: 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba
    vcpkg-scripts version: 4cb4a5c5d 2022-11-18 (8 hours ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Using cached readline-8.1.tar.gz.
-- Cleaning sources at /data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/src/readline-8-607d72e973.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /data/sasr/pkg_manager/vcpkg/downloads/readline-8.1.tar.gz
-- Using source at /data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/src/readline-8-607d72e973.clean
-- Getting CMake variables for arm92-buildroot-linux-dynamic-dbg
-- Getting CMake variables for arm92-buildroot-linux-dynamic-rel
-- Configuring arm92-buildroot-linux-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: /usr/bin/bash -c "V=1 ./../src/readline-8-607d72e973.clean/configure  \"--with-curses=yes\" \"--disable-silent-rules\" \"--verbose\" \"--enable-shared\" \"--disable-static\" \"--prefix=/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug\" \"--bindir=\\${prefix}/../tools/readline-unix/debug/bin\" \"--sbindir=\\${prefix}/../tools/readline-unix/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/readline-unix\""
    Working Directory: /data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/arm92-buildroot-linux-dynamic-dbg
    Error code: 1
    See logs for more information:
      /data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/config-arm92-buildroot-linux-dynamic-dbg-config.log
      /data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/config-arm92-buildroot-linux-dynamic-dbg-out.log
      /data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/config-arm92-buildroot-linux-dynamic-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:808 (vcpkg_execute_required_process)
  buildtrees/versioning_/versions/readline-unix/e81a9fe4baa7c437b6e6d9636ae7c291e2ed81e9/portfile.cmake:10 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)



```
<details><summary>/data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/config-arm92-buildroot-linux-dynamic-dbg-config.log</summary>

```
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by readline configure 8.1, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ ./../src/readline-8-607d72e973.clean/configure --with-curses=yes --disable-silent-rules --verbose --enable-shared --disable-static --prefix=/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug --bindir=${prefix}/../tools/readline-unix/debug/bin --sbindir=${prefix}/../tools/readline-unix/debug/sbin --libdir=${prefix}/lib --includedir=${prefix}/../include --datarootdir=${prefix}/share/readline-unix

## --------- ##
## Platform. ##
## --------- ##

hostname = UIRSCSRPBULDW2V
uname -m = x86_64
uname -r = 5.15.0-52-generic
uname -s = Linux
uname -v = #58-Ubuntu SMP Thu Oct 13 08:03:55 UTC 2022

/usr/bin/uname -p = x86_64
/bin/uname -X     = unknown

/bin/arch              = x86_64
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = unknown
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /home/sasr/.vscode-server/bin/b06ae3b2d2dbfe28bca3134cc6be65935cdfea6a/bin/remote-cli
PATH: /usr/local/sbin
PATH: /usr/local/bin
PATH: /usr/sbin
PATH: /usr/bin
PATH: /sbin
PATH: /bin
PATH: /usr/games
PATH: /usr/local/games
PATH: /snap/bin
PATH: /data/sasr/pkg_manager/vcpkg/downloads/tools/ninja/1.10.2-linux
PATH: /bin


## ----------- ##
## Core tests. ##
## ----------- ##

configure:2490: checking build system type
configure:2504: result: x86_64-pc-linux-gnu
configure:2524: checking host system type
configure:2537: result: x86_64-pc-linux-gnu
configure:2650: checking whether make sets $(MAKE)
configure:2672: result: yes
configure:2729: checking for gcc
configure:2756: result: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc
configure:2985: checking for C compiler version
configure:2994: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc --version >&5
arm-buildroot-linux-gnueabi-gcc.br_real (Buildroot 2019.11.1) 9.2.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:3005: $? = 0
configure:2994: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc -v >&5
Using built-in specs.
COLLECT_GCC=/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc.br_real
COLLECT_LTO_WRAPPER=/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/../libexec/gcc/arm-buildroot-linux-gnueabi/9.2.0/lto-wrapper
Target: arm-buildroot-linux-gnueabi
Configured with: ./configure --prefix=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32 --sysconfdir=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/etc --enable-static --target=arm-buildroot-linux-gnueabi --with-sysroot=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/arm-buildroot-linux-gnueabi/sysroot --enable-__cxa_atexit --with-gnu-ld --disable-libssp --disable-multilib --disable-decimal-float --with-gmp=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32 --with-mpc=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32 --with-mpfr=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32 --with-pkgversion='Buildroot 2019.11.1' --with-bugurl=http://bugs.buildroot.net/ --disable-libquadmath --enable-tls --enable-threads --with-isl=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32 --with-float=soft --with-abi=aapcs-linux --with-cpu=cortex-a9 --with-float=soft --with-mode=arm --enable-languages=c,c++ --with-build-time-tools=/opt/toolchains/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/arm-buildroot-linux-gnueabi/bin --enable-shared --disable-libgomp
Thread model: posix
gcc version 9.2.0 (Buildroot 2019.11.1) 
configure:3005: $? = 0
configure:2994: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc -V >&5
arm-buildroot-linux-gnueabi-gcc.br_real: error: unrecognized command line option '-V'
arm-buildroot-linux-gnueabi-gcc.br_real: fatal error: no input files
compilation terminated.
configure:3005: $? = 1
configure:2994: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc -qversion >&5
arm-buildroot-linux-gnueabi-gcc.br_real: error: unrecognized command line option '-qversion'; did you mean '--version'?
arm-buildroot-linux-gnueabi-gcc.br_real: fatal error: no input files
compilation terminated.
configure:3005: $? = 1
configure:3025: checking whether the C compiler works
configure:3047: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc -g  -L/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug/lib conftest.c  >&5
configure:3051: $? = 0
configure:3099: result: yes
configure:3102: checking for C compiler default output file name
configure:3104: result: a.out
configure:3110: checking for suffix of executables
configure:3117: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc -o conftest -g  -L/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug/lib conftest.c  >&5
configure:3121: $? = 0
configure:3143: result: 
configure:3165: checking whether we are cross compiling
configure:3173: /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc -o conftest -g  -L/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug/lib conftest.c  >&5
configure:3177: $? = 0
configure:3184: ./conftest
./../src/readline-8-607d72e973.clean/configure: line 3186: ./conftest: cannot execute binary file: Exec format error
configure:3188: $? = 126
configure:3195: error: in `/data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/arm92-buildroot-linux-dynamic-dbg':
configure:3197: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_build=x86_64-pc-linux-gnu
ac_cv_env_CC_set=set
ac_cv_env_CC_value=/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc
ac_cv_env_CFLAGS_set=set
ac_cv_env_CFLAGS_value=-g
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_CPP_set=
ac_cv_env_CPP_value=
ac_cv_env_LDFLAGS_set=set
ac_cv_env_LDFLAGS_value=-L/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug/lib
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_host_alias_set=
ac_cv_env_host_alias_value=
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_host=x86_64-pc-linux-gnu
ac_cv_prog_ac_ct_CC=/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc
ac_cv_prog_make_make_set=yes

## ----------------- ##
## Output variables. ##
## ----------------- ##

AR='/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-ar'
ARFLAGS=''
BRACKETED_PASTE='-DBRACKETED_PASTE_DEFAULT=1'
BUILD_DIR=''
CC='/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc'
CFLAGS='-g'
CPP=''
CPPFLAGS=''
CROSS_COMPILE=''
DEFS=''
ECHO_C=''
ECHO_N='-n'
ECHO_T=''
EGREP=''
EXAMPLES_INSTALL_TARGET=''
EXEEXT=''
GREP=''
INSTALL_DATA=''
INSTALL_PROGRAM=''
INSTALL_SCRIPT=''
LDFLAGS='-L/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug/lib'
LIBOBJS=''
LIBS=''
LIBVERSION='8.1'
LOCAL_CFLAGS=''
LOCAL_DEFS=''
LOCAL_LDFLAGS=''
LTLIBOBJS=''
MAKE_SHELL=''
OBJEXT=''
PACKAGE_BUGREPORT='bug-readline@gnu.org'
PACKAGE_NAME='readline'
PACKAGE_STRING='readline 8.1'
PACKAGE_TARNAME='readline'
PACKAGE_URL=''
PACKAGE_VERSION='8.1'
PATH_SEPARATOR=':'
RANLIB='/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-ranlib'
SET_MAKE=''
SHARED_INSTALL_TARGET=''
SHARED_TARGET=''
SHELL='/bin/bash'
SHLIB_DLLVERSION=''
SHLIB_DOT=''
SHLIB_LIBPREF=''
SHLIB_LIBS=''
SHLIB_LIBSUFF=''
SHLIB_LIBVERSION=''
SHLIB_MAJOR=''
SHLIB_MINOR=''
SHLIB_STATUS=''
SHLIB_XLDFLAGS=''
SHOBJ_CC=''
SHOBJ_CFLAGS=''
SHOBJ_LD=''
SHOBJ_LDFLAGS=''
SHOBJ_LIBS=''
SHOBJ_STATUS=''
SHOBJ_XLDFLAGS=''
STATIC_INSTALL_TARGET=''
STATIC_TARGET=''
TERMCAP_LIB=''
TERMCAP_PKG_CONFIG_LIB=''
ac_ct_CC='/opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc'
bindir='${prefix}/../tools/readline-unix/debug/bin'
build='x86_64-pc-linux-gnu'
build_alias=''
build_cpu='x86_64'
build_os='linux-gnu'
build_vendor='pc'
datadir='${datarootdir}'
datarootdir='${prefix}/share/readline-unix'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
exec_prefix='NONE'
host='x86_64-pc-linux-gnu'
host_alias=''
host_cpu='x86_64'
host_os='linux-gnu'
host_vendor='pc'
htmldir='${docdir}'
includedir='${prefix}/../include'
infodir='${datarootdir}/info'
libdir='${prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/data/sasr/gitstash/crsp-sdk-dependencies/vcpkg/bhrx/build/vcpkg_installed/arm92-buildroot-linux-dynamic/debug'
program_transform_name='s,x,x,'
psdir='${docdir}'
sbindir='${prefix}/../tools/readline-unix/debug/sbin'
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target_alias=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "readline"
#define PACKAGE_TARNAME "readline"
#define PACKAGE_VERSION "8.1"
#define PACKAGE_STRING "readline 8.1"
#define PACKAGE_BUGREPORT "bug-readline@gnu.org"
#define PACKAGE_URL ""

configure: exit 1
```
</details>
<details><summary>/data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/config-arm92-buildroot-linux-dynamic-dbg-out.log</summary>

```
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu

Beginning configuration for readline-8.1 for x86_64-pc-linux-gnu

checking whether make sets $(MAKE)... yes
checking for gcc... /opt/toolchain/crosstools-arm-gcc-9.2-linux-4.19-glibc-2.30-binutils-2.32/bin/arm-buildroot-linux-gnueabi-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... 
```
</details>
<details><summary>/data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/config-arm92-buildroot-linux-dynamic-dbg-err.log</summary>

```
configure: WARNING: unrecognized options: --disable-silent-rules
configure: error: in `/data/sasr/pkg_manager/vcpkg/buildtrees/readline-unix/arm92-buildroot-linux-dynamic-dbg':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "name": "vzsdk-bhrx1-depends",
  "version-string": "1.0.0",
  "port-version": 0,
  "description": "BHRx1 third party dependencies",
  "dependencies": [
    "openssl",
    "zlib",
    "libuuid",
    "jansson",
    "fastcgi",
    "curl",
    "paho-mqtt",
    "libzip",
    "pcre",
    "pcre2",
    "benchmark",
    "gtest",
    "breakpad",
    "ncurses",
    "readline"
  ],
  "builtin-baseline": "4cb4a5c5ddcb9de0c83c85837ee6974c8333f032"
}

```
</details>


**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

Built it on the manifest mode by using overlay triplets:

```
{
  "name": "depends",
  "version-string": "1.0.0",
  "port-version": 0,
  "description": "third party dependencies",
  "dependencies": [
    "openssl",
    "zlib",
    "libuuid",
    "jansson",
    "fastcgi",
    "curl",
    "paho-mqtt",
    "libzip",
    "pcre",
    "pcre2",
    "benchmark",
    "gtest",
    "breakpad",
    "ncurses",
    "readline"
  ],
  "builtin-baseline": "4cb4a5c5ddcb9de0c83c85837ee6974c8333f032"
}
```

--- a/ports/readline-unix/portfile.cmake
+++ b/ports/readline-unix/portfile.cmake
@@ -1,22 +1,27 @@
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
+set(filename readline-${VERSION}.tar.gz)
 vcpkg_download_distfile(
-  ARCHIVE
-  URLS https://ftp.gnu.org/gnu/readline/readline-8.1.tar.gz
-  FILENAME readline-8.1.tar.gz
-  SHA512 27790d0461da3093a7fee6e89a51dcab5dc61928ec42e9228ab36493b17220641d5e481ea3d8fee5ee0044c70bf960f55c7d3f1a704cf6b9c42e5c269b797e00
+    ARCHIVE
+    URLS
+        "https://ftpmirror.gnu.org/gnu/readline/${filename}"
+        "https://ftp.gnu.org/gnu/readline/${filename}"
+    FILENAME "${filename}"
+    SHA512 0a451d459146bfdeecc9cdd94bda6a6416d3e93abd80885a40b334312f16eb890f8618a27ca26868cebbddf1224983e631b1cbc002c1a4d1cd0d65fba9fea49a
 )
 
 vcpkg_extract_source_archive(SOURCE_PATH ARCHIVE "${ARCHIVE}")
 
 vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
+    DETERMINE_BUILD_TRIPLET
     OPTIONS
         --with-curses=yes
+        --disable-install-examples
 )
 
 vcpkg_install_make()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/tools/readline-unix/bin" "${CURRENT_PACKAGES_DIR}/tools/readline-unix/debug/bin")
 
 vcpkg_fixup_pkgconfig()
 

--- a/ports/readline-unix/vcpkg.json
+++ b/ports/readline-unix/vcpkg.json
@@ -1,8 +1,8 @@
 {
   "name": "readline-unix",
-  "version": "8.1",
-  "port-version": 2,
-  "description": "Implementation of readline for unix",
+  "version": "8.2",
+  "description": "The GNU Readline library provides a set of functions for use by applications that allow users to edit command lines as they are typed in.",
+  "homepage": "https://tiswww.case.edu/php/chet/readline/rltop.html",
   "license": "GPL-3.0-or-later",
   "supports": "!windows",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6529,8 +6529,8 @@
       "port-version": 0
     },
     "readline-unix": {
-      "baseline": "8.1",
-      "port-version": 2
+      "baseline": "8.2",
+      "port-version": 0
     },
     "readline-win32": {
       "baseline": "5.0",

--- a/versions/r-/readline-unix.json
+++ b/versions/r-/readline-unix.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "19fcd2b764ea6d261dfbd80961833e807cf93ea0",
+      "version": "8.2",
+      "port-version": 0
+    },
+    {
       "git-tree": "e81a9fe4baa7c437b6e6d9636ae7c291e2ed81e9",
       "version": "8.1",
       "port-version": 2

